src/client/Pipeline.cpp

```diff
@@ -880,6 +907,7 @@ void PipelineImpl::flush() {
 }
 
 void PipelineImpl::waitForAcks(bool force) {
+    lock_guard < mutex > lock(mut);
```

Comment:
Is this lock scope too large? There is a loop in it, may cause some performance issue